Automattic/wp-calypso
# Jetpack Backup: Add Backup retention management setting

#### Proposed Changes

* Add a new `Days of backups saved` (backup retention period) setting in Jetpack Cloud, under Settings page. For this PR, it will be available only for A12s (dev & stage environment). The idea is to start the review process of this new functionality.
* This iteration includes:
  * Allow to update the setting only to a number of days that doesn't require a storage upgrade.
  * Fetch and update the customer-defined retention period to our backend.
  * Confirmation dialog when attempting to update to a lower retention period.

#### Demo

https://user-images.githubusercontent.com/1488641/218225112-ce96783e-d079-447b-8df2-24d9f54666fd.mov

↪ [Demo v1 (deprecated)](https://user-images.githubusercontent.com/1488641/217998763-cc89187c-9ce8-4536-a8fb-785c153ab76c.mov)


#### Screenshots

| Description | Screenshot |
|---|---|
| Setting page | ![image](https://user-images.githubusercontent.com/1488641/218224377-49cd64f7-bc66-4748-a0fa-999049c8e638.png) |
| Confirmation dialog | ![image](https://user-images.githubusercontent.com/1488641/218224478-25866771-8b23-49ee-9094-441fa6a1960a.png) |
| Tooltip | ![image](https://user-images.githubusercontent.com/1488641/218224599-6b25c3cb-d23b-46f2-a851-c97d086dfe33.png) |
| Success notice | ![image](https://user-images.githubusercontent.com/1488641/218224332-d387d0ab-e73e-4d3b-9377-54dba3f57fab.png) |
| Error notice | ![image](https://user-images.githubusercontent.com/1488641/218000224-10eb567e-2f3c-4e94-8613-c1f2947d8328.png) |




#### Testing Instructions



* Start a Jetpack Cloud live branch
* Navigate to Settings and wait for the new setting to load.
* The `Update settings` button will be disabled when the current plan or an option that requires an upgrade are selected.
* You could try to update the days of backups saved. The way to re-confirm it was properly saved is refreshing the page.

Notes:
* If you haven't configured this setting before, the current retention period associated to your plan will be preselected. Otherwise, your current configuration will be preselected.

#### Pre-merge Checklist
